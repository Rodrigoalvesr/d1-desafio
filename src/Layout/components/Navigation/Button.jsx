import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@Components/Tooltip';
import Icon from '@Components/Icon';
import { makeStyles } from '@material-ui/core/styles';
import classnames from 'classnames';

const useStyles = makeStyles((theme) => ({
  button: {
    display: 'flex',
    justifyContent: 'center',
  },
  content: {
    width: 21,
    padding: '24px 0',
    '&.selected': {
      borderWidth: '0 0 1px 0',
      borderStyle: 'solid',
      borderImage: 'linear-gradient(90deg, #00E1FF 0%, #117EFF 100%) 1',
      boxSizing: 'border-box',
    },
  },
  icon: {
    cursor: 'pointer',
    fill: '#E4E6F1',
    '&:hover': {
      fill: theme.palette.primary.main,
    },
  },
}));

function Button({ name, icon, selected }) {
  const classes = useStyles();

  return (
    <div className={classes.button}>
      <div className={classnames(classes.content, { selected })}>
        <Tooltip title={name} placement="right" arrow>
          <div>
            <Icon name={icon} className={classes.icon} />
          </div>
        </Tooltip>
      </div>
    </div>
  );
}

Button.defaultProps = {
  selected: false,
};

Button.propTypes = {
  name: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  selected: PropTypes.bool,
};

export default Button;
