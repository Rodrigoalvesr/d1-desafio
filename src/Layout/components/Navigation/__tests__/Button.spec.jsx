import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
  it('Renders correctly', () => {
    const component = renderer
      .create(<Button name="AnÃ¡lises" icon="chart" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
