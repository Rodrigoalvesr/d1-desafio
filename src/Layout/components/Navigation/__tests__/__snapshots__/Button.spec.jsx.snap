// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders correctly 1`] = `
<div
  className="makeStyles-button-1"
>
  <div
    className="makeStyles-content-2"
  >
    <div
      aria-describedby={null}
      className=""
      onBlur={[Function]}
      onFocus={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchStart={[Function]}
      title="AnÃ¡lises"
    >
      <svg
        className="makeStyles-root-6 makeStyles-root-7 makeStyles-icon-3"
      >
        chart-pie.svg
      </svg>
    </div>
  </div>
</div>
`;
